### DISCUSSION TITLE
Best way to get a (new) workplane within a Shape

### DISCUSSION THREAD

**Discussion by Jopie01** [CadQuery/cadquery#1642] [Q&A]
Maybe a very confusing title 😓  but what I mean is that I have extruded a profile. This profile is imported from a DXF file and placed on the "XZ" workplane. Now I want to cut a piece from it but I don't want to use a face of the profile because the profile can change shape. Instead of using a face I want to use the "YZ" workplane which should be the local one and not the global one.

According to the documentation I can use `copyWorkplane` to get a new workplane which is working properly, but I was wondering it that's the correct way to do this. Maybe there are other ways as well?

**Comment by adam-urbanczyk** [comment]
Are you looking for `cq.Workplane.transformed`? Without an example, It is rather difficult to understand what you are after.

**Comment by Jopie01** [comment]
> Without an example, It is rather difficult to understand what you are after.

You are right, I've put an example together.

```python
import cadquery as cq

# first I create a beam
beam = (cq.Workplane("XZ")
    .rect(50, 100)
    .extrude(1000)
)

# now I want to cut a slot in the beam but without doing `faces(">X").workplane()`
# I want the local YZ workplane instead
result = (beam.copyWorkplane(cq.Workplane("YZ", origin=(0, -500, 0)))
    .rect(200, 50)
    .extrude(50, combine='cut')
)

show_object(result)
```
Hopefully this clears things up a bit. 

The question is if this is the right thing to do or is there a better approach. It's basically the same when doing this in a GUI. Maybe `transformed` can be used, but then also the local origin is modified which I don't want.

The reason for not using faces is because the shape of the profile can be rounded and dynamically changed. Selecting the right face becomes quite difficult hence using the workplane instead.

**Comment by adam-urbanczyk** [comment]
I do not understand what is the problem with `faces(">X").workplane()`. If you extrude a planar profile you can always select the right face. You remark about changing local origin does not make sense to me either, copy workplane will also modify it. What is the problem at hand?

Maybe you want to use the free function API? Maybe you want to define a custom selector?

**Comment by Jopie01** [comment]
The example is just an example with a rectangle, but maybe a circle was better. In the function I create, the beam profile is created by an imported DXF and can have any form and size.

> I do not understand what is the problem with `faces(">X").workplane()`

When using faces I'm at the mercy of CadQuery to select the right face. Also like I said the faces are maybe planer. And if they are planar they can be rotated around an axis which ruins the cutout.

> You remark about changing local origin does not make sense to me either, copy workplane will also modify it.

Sorry, your are right, the example is wrong, I used `moveTo` to keep the origin at the same place.

> What is the problem at hand?

I want to make a cut in a beam or cut the beam at an angle. The profile of the beam is created by importing a DXF file, so the profile of the beam is kind of unknown beforehand. So I want to have a geometry independent sketch to be used to make an angular cut at the end of the beam. Also a cutout at a certain place will be added. Another thing is that the origin (0, 0, 0) should not change because I'm creating an assembly based on the origins.

Using the `copyWorkplane` in combination with `moveTo` works very well in this case, but I was wondering if there are other ways to do this. For the use of `copyWorkplane` I read https://cadquery.readthedocs.io/en/latest/examples.html#copying-workplanes
